frontend/controllers/product-controller.ts

```diff
@@ -40,6 +43,11 @@ export default class ShopifyBuyControllerBase extends Controller {
     this.getCart();
   }
 
+  updateAddToCartButton() {
+    const selectedVariantId = this.variantSelectorTarget.value;
+    this.addToCartButtonTarget.setAttribute('data-product-variant-id-param', selectedVariantId);
+  }
```

Comment:
this may need to change the `disabled` attribute on the button:

`{% if product.variants.0.inventory_quantity == 0 and product.variants.0.inventory_policy == 'deny' %} disabled="true"`

Reply:
thanks for catching that. I think I've handled that now - could you check?